### PR TITLE
fix: use the new matrix permissions syntax introduced with matrix-auth v3.0

### PR DESCRIPTION
**Pull request by lemeurherve**

### CODE REVIEW

File: config/jenkins-release.yaml [hunk 1]
```diff
@@ -479,10 +479,12 @@ controller:
           authorizationStrategy:
             globalMatrix:
               permissions:
-                - "Overall/Administer:release-core"
-                - "Overall/SystemRead:authenticated"
-                - "Overall/Read:authenticated"
-                - "Job/Read:authenticated"
+                - "GROUP:Overall/Administer:release-core"
+                - "GROUP:Overall/Administer:admins"
```
Comment (lemeurherve) [inline config/jenkins-release.yaml:483]: @daniel-beck @timja in common with @dduportal we've added this permission for our "admin" group on release.ci.jenkins.io, as we have already admin rights on other services.
OK for you?

File: config/jenkins-release.yaml [hunk 2]
```diff
@@ -479,10 +479,12 @@ controller:
...
+                - "GROUP:Overall/Administer:jenkins-admins"
```
Comment (lemeurherve) [inline config/jenkins-release.yaml:484]: @daniel-beck @timja again in common with @dduportal we were wondering if the "jenkins-admin" group should be added to release.ci.jenkins.io.
WDYT?

File: config/jenkins-release.yaml (hunk 1 above)
Comment (daniel-beck) [reply config/jenkins-release.yaml:483]: This isn't cert.ci so up to infra team 🤷

**Review by timja — APPROVED** [review]
fine with me, the people are generally the same anyway, tbh the general release team doesn't really need admin anyway just build.

**Review by dduportal — APPROVED** [review]
👍

**Pull request merged**